build.sbt

```diff
@@ -1335,18 +1335,51 @@ lazy val scalaStyleSettings = Seq(
  ****************************
  */
 
+lazy val compileJavastyle = taskKey[Unit]("compileJavastyle")
+lazy val testJavastyle = taskKey[Unit]("testJavastyle")
+
 def javaCheckstyleSettings(checkstyleFile: String): Def.SettingsDefinition = {
   // Can be run explicitly via: build/sbt $module/checkstyle
   // Will automatically be run during compilation (e.g. build/sbt compile)
   // and during tests (e.g. build/sbt test)
   Seq(
     checkstyleConfigLocation := CheckstyleConfigLocation.File(checkstyleFile),
-    checkstyleSeverityLevel := CheckstyleSeverityLevel.Error,
-    (Compile / compile) := ((Compile / compile) dependsOn (Compile / checkstyle)).value,
-    (Test / test) := ((Test / test) dependsOn (Test / checkstyle)).value
+    checkstyleSeverityLevel := CheckstyleSeverityLevel.Ignore,
```

Comment:
one naive qn: why is this changed from `Error` to `Ignore`?

Reply:
> Is it because we want to control the part about throwing error?

@vkorukanti Yes, that's right. If we keep the `Error` severity, `build/sbt` will throw an error and immediately stop at the checkstyle phase (if error) -> never execute the check-report phase of `checkstyle-report.xml` and `checkstyle-test-report.xml`. We need to `Ignore` and throw error if exists when checking `*report.xml`.